x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -112,19 +114,35 @@ export function createExecutionHandler<
         }),
       }));
 
-    alertExecutionStore.numberOfScheduledActions += actions.length;
+    alertExecutionStore.incrementNumberOfScheduledActions(actions.length);
 
     const ruleLabel = `${ruleType.id}:${ruleId}: '${ruleName}'`;
 
     const actionsClient = await actionsPlugin.getActionsClientWithRequest(request);
     let ephemeralActionsToSchedule = maxEphemeralActionsPerRule;
 
     for (const action of actions) {
-      if (alertExecutionStore.numberOfTriggeredActions >= ruleType.config!.execution.actions.max) {
-        alertExecutionStore.triggeredActionsStatus = ActionsCompletion.PARTIAL;
+      if (alertExecutionStore.hasReachedTheExecutableActionsLimit(actionsConfigMap)) {
+        alertExecutionStore.setTriggeredActionsStatus(ActionsCompletion.PARTIAL);
+        logger.debug(
+          `The maximum number of actions (${actionsConfigMap.default.max}) for this rule type has been reached`
+        );
         break;
       }
 
+      if (
+        alertExecutionStore.hasReachedTheExecutableActionsLimitByConnectorType({
+          actionTypeId: action.actionTypeId,
+          actionsConfigMap,
+        })
+      ) {
+        alertExecutionStore.setTriggeredActionsStatus(ActionsCompletion.PARTIAL);
+        logger.debug(
+          `The maximum number of actions for this connector type (${action.actionTypeId}) type has been reached`
```

Comment:
Unlike the check ^^^, this one continues, so the message will be repeated for every action of this type triggered afterwards (for this rule execution).  Feels like that will get super noisy when this goes off.  Maybe we need to keep a map of the connector type and whether or not it's gone off, to see if we need to log.

Reply:
I removed the logger completely. Maybe we can add it with a follow-up issue later.